sdk/core/core/test/cmocka/test_az_encode.c

```diff
@@ -63,3 +63,11 @@ void test_url_encode(void** state)
     assert_true(az_span_is_content_equal(builder, uri_encoded));
   }
 }
+
+int test_az_encode()
```

Comment:
I prefer to keep this as `test_url_encode` and keep the original file name test_url_encode. We are't testing anything other than url_encode here.

In fact, we may want to move the tests to az_span because that is the API we are testing here (and remove this fie).